File: Perl/perm-missing-elem.pl
#! /usr/bin/perl

use strict;
use warnings;

# http://codility.com/demo/take-sample-test/permmissingelem

sub solution {
    my (@a)=@_;
    @a = sort @a;
    my $al = (0 + @a);
    # print ">@a< \n";
    # print $al."\n";
    if ( $al == 0) {
        return 1;
    } else {
        foreach my $x (0..$al-1) {
            unless ($a[$x] == $x+1) {
                return $x+1;
            }
        }
        return $al+1
    }
}

print "\n";
print solution(3,2,1,5);
print "\n";
print solution();
print "\n";
print solution(1);
print "\n";
print solution(1,2);

print "\n";
print (3 == 2);

File: Ruby/Codility/Lesson-1/perm-missing-elem.rb
def solution(a)
  a.sort!
  n = a.size
  #p a
  if n == 0
    return 1
  elsif n == 1
    if a[0] == 1
      return a[0]+1
    else
      return 1
    end
  else
    for x in 0..(n-1)
      unless a[x] == x+1
        return x+1
      end
    end
  end
  n+1
end


p solution [2, 3, 1, 5]
p solution [3, 1]
p solution [2, 1]
p solution [1]
p solution [2] #must return 1
p solution []
